### ISSUE TITLE
Document regex examples of popular selector naming conventions (camelcase, kebabcase etc)

### ISSUE THREAD

**Issue by Awk34** [stylelint/stylelint#1951]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I added the following rule to my `.stylelintrc` to enforce kebab-case selector names: `"selector-class-pattern": "^([a-z][a-z0-9]*)(-[a-z0-9]+)*$"`. It enforces that a class starts with at least one lowercase letter character + 0 or more lowercase letters/numbers, and is followed by 0 or more dashes + any mix of lowercase letters/numbers. Is this something that could maybe be added to this project? Or at least to a doc file? There are probably a lot of people like me whom want to enforce such class names, but maybe aren't great at RegEx. There could maybe be shorthands for the rule like `kebab-case` or `camelCase`

> Which rule, if any, is this issue related to?

`selector-class-pattern`

> What CSS is needed to reproduce this issue?

Good
```css
.foo {}
.foo-bar1 {}
.a-10
```
Bad
```css
.Foo {}
.foo-bar1- {}
.1f {}
```

> What stylelint configuration is needed to reproduce this issue?

(noted above)

> Which version of stylelint are you using?

`^6.5.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Gulp + PostCSS plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no

> What did you expect to happen?

N/A

> What actually happened (e.g. what warnings or errors you are getting)?

N/A

**Comment by jeddy3** [comment]
@Awk34 Thanks for the suggestion.

> Or at least to a doc file?

I think this could be a useful entry in the FAQ. We could have the regex for:

- kebab case
- snake case
- camel case
- pascal case

As a quick google doesn't seem to return any easy-to-find reference guides.

**Comment by evilebottnawi** [comment]
@jeddy3 sgtm, i wanted to offer too

**Comment by Nirazul** [comment]
I'd also add a regex pattern for the BEM naming convention as it is widely used out in the wild.

**Comment by davidtheclark** [comment]
https://github.com/postcss/postcss-bem-linter should have some examples to start with.

**Comment by jeddy3** [comment]
@Nirazul I didn't jot down BEM naming conventions as, like @davidtheclark said, I assumed people would be using `postcss-bem-linter` via [`stylelint-selector-bem-pattern`](https://github.com/davidtheclark/stylelint-selector-bem-pattern) if they're using BEM conventions.

We can certainly add a line about `stylelint-selector-bem-pattern` to the FAQ entry e.g. "Use `stylelint-selector-bem-pattern` to enforce various flavours of BEM naming conventions."

Unless anyone has any objections, I'll label this up as a documentation issue and requiring help.
